### ISSUE TITLE
Tensorflow Estimator passes train data through some weird normalization before entering net

### ISSUE THREAD

**Issue by yonatansc97** [tensorflow/tensorflow#45428]
**System information**
- Have I written custom code: Yes
- OS Platform and Distribution: Linux Ubuntu 18.04
- TensorFlow installed from: anaconda
- TensorFlow version: 2.2, but some parts of the networks use tf.compat.v1
- Python version: 3

**Describe the current behavior**

I am using tensorflow Estimator API, and have encountered a weird phenomenon. I am passing the exact same input_fn to both training and evaluation, and for some reason the images which are provided to the network are not identical. I use the tf.Estimator.train_and_evaluate, and use tensorboard to display the images that are passed to the model_fn. This means that these images are those that are later fed into the network, and are not an intermediate result from the network. The two images seem similar, but after taking a closer look, it seems that evaluation images are ok, but train images are somewhat distorted.

After loading them both, I noticed that for some reason the training images go through some kind of ReLu. I affirmed it with this code:

special_relu = lambda mat: ((mat - 0.5) / 0.5) * ((mat - 0.5) / 0.5 > 0)
np.allclose(mat_train, special_relu(mat_eval))
>>> True

**Describe the expected behavior**

The expected behaviour is that the evaluation images should be the same as the train images, such that:
np.allclose(mat_train, mat_eval)
>>> True

Clarification: The input_fn I provided is one that yields the exact same image over and over again, so the images that are drawn should all be the same, theoretically.

**What I thought and tried**

My initial thought was that it is some form of BatchNormalization. But BatchNormalization is supposed to happen within the network, and not as some preprocess, shouldn't it? What I recorded (using tf.summary.image) was the features['image'] object, passed to my model_fn. And if I understand correctly, the features object is passed to model_fn by the input_fn called by the Estimator object.

Regardless, I tried to remove the parts in the code which are supposed to call the BatchNormalization. This had no effect. Of course, I might have not done that in the right way, but as I said it I don't really think it is BatchNormalization.

**Comment by ravikyram** [comment]
@yonatansc97 

Please, share colab link or simple standalone code with supporting files to reproduce the issue in our environment. It helps us in localizing the issue faster. Thanks!

**Comment by yonatansc97** [comment]
Hi, @ravikyram
Thank you for the response!
I could not get it to run on a colab environment, so I am uploading the following files of code:
- inception_v3.py - this is the CNN.
- modeling.py - this module implements the tf.Estimator API, and is responsible for handling tensorboard summaries (those that are not taken care of by the Estimator itself).

- main.py - this creates the estimator and runs it, with different input functions that you can play with. I implemented a couple of input functions which generate fake images in various patterns ('steps', 'zebra', 'arange', 'arange_mod_10') and another input function that loads an image similar to the actual scenario ('load_img'). You can pick and choose the input_fn to use by changing the variable input_fn_mode to any of the above.

I am also attaching the two files:
- my_img.npy - the image which I load when the input_fn_mode='load_img'
- environment.yml - my conda environment

The behaviour I mentioned (with the special relu) happens on the 'load_img' input_fn, and in the other input_fn_mode's other weird things happen, which make the eval images look different than train images.

[all_files.zip](https://github.com/tensorflow/tensorflow/files/5651109/all_files.zip)

Should any additional information/clarification be needed, please let me know.
Thanks in advance!

**Comment by sushreebarsa** [comment]
@yonatansc97 It looks like you are using an older Version of Tensorflow. Many bugs have been fixed in the latest version. Could you please execute your code using Latest  stable Version of TF 2.5 and let us know if the issue still persists? Thanks!

**Comment by yonatansc97** [comment]
Actually, I have since then moved to pytorch. No such problems now, I'm afraid.
